xds/internal/client/client_xds.go

```diff
@@ -382,6 +384,32 @@ func securityConfigFromCluster(cluster *v3clusterpb.Cluster) (*SecurityConfig, e
 	return sc, nil
 }
 
+// circuitBreakersFromCluster extracts the circuit breakers configuration from
+// the received cluster resource. Returns nil if no CircuitBreakers or no
+// Thresholds in CircuitBreakers.
+func circuitBreakersFromCluster(cluster *v3clusterpb.Cluster) *uint32 {
+	circuitBreakers := cluster.GetCircuitBreakers()
+	if circuitBreakers == nil {
+		return nil
+	}
+	thresholds := circuitBreakers.GetThresholds()
+	if thresholds == nil {
+		return nil
+	}
+	for _, threshold := range thresholds {
+		if threshold.GetPriority().String() != "DEFAULT" {
+			continue
+		}
+		maxRequestsPb := threshold.GetMaxRequests()
+		var maxRequests uint32 = 1024
```

Comment:
Just `return nil` if `MaxRequests` is not set in `DEFAULT`.

Reply:
Made this change.